File: Core.Tests/FormatTableTests.fs
namespace Extensions

open System
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type FormatTableTests() =
    [<TestMethod>]
    member _.``format table`` () =
        let mutable table = FormatTable(5, Alignment.Left)
        table.Format("one", "two", "three", "four", "five")
        Assert.AreEqual("one│two│three│four│five\n", table.ToString())
        table.Format("six", "seven", "eight", "nine", "ten")
        Assert.AreEqual("one│two  │three│four│five\nsix│seven│eight│nine│ten \n", table.ToString())
        table <- FormatTable(5, Alignment.Right)
        table.Format("one", "two", "three", "four", "five")
        Assert.AreEqual("one│two│three│four│five\n", table.ToString())
        table.Format("six", "seven", "eight", "nine", "ten")
        Assert.AreEqual("one│  two│three│four│five\nsix│seven│eight│nine│ ten\n", table.ToString())
        table <- FormatTable(5, Alignment.Center)
        table.Format("one", "two", "three", "four", "five")
        Assert.AreEqual("one│two│three│four│five\n", table.ToString())
        table.Format("six", "seven", "eight", "nine", "ten")
        Assert.AreEqual("one│ two │three│four│five\nsix│seven│eight│nine│ten \n", table.ToString())

File: Tests/FormatTableTests.cs
using System;
using Stringier;
using Xunit;

// This file contains many calls to .Replace("\r\n", "\n") throughout it, they are present so that these tests will work across operating systems, which have different conventions for line terminator sequences.

namespace Tests {
	public class FormatTableTests {

		[Theory]
		[InlineData(new[] { "one", "two", "three", "four", "five" }, "one│two│three│four│five\n", new[] { "six", "seven", "eight", "nine", "ten" }, "one│two  │three│four│five\nsix│seven│eight│nine│ten \n")]
		public void FormatTable_AlignLeft(String[] firstFormatParams, String firstExpected, String[] secondFormatParams, String secondExpected) {
			FormatTable table = new FormatTable(5, Alignment.Left);
			table.Format(firstFormatParams);
			Assert.Equal(firstExpected, table.ToString().Replace("\r\n", "\n"));
			table.Format(secondFormatParams);
			Assert.Equal(secondExpected, table.ToString().Replace("\r\n", "\n"));
		}

		[Theory]
		[InlineData(new[] { "one", "two", "three", "four", "five" }, "one│two│three│four│five\n", new[] { "six", "seven", "eight", "nine", "ten" }, "one│  two│three│four│five\nsix│seven│eight│nine│ ten\n")]
		public void FormatTable_AlignRight(String[] firstFormatParams, String firstExpected, String[] secondFormatParams, String secondExpected) {
			FormatTable table = new FormatTable(5, Alignment.Right);
			table.Format(firstFormatParams);
			Assert.Equal(firstExpected, table.ToString().Replace("\r\n", "\n"));
			table.Format(secondFormatParams);
			Assert.Equal(secondExpected, table.ToString().Replace("\r\n", "\n"));
		}

		[Theory]
		[InlineData(new[] { "one", "two", "three", "four", "five" }, "one│two│three│four│five\n", new[] { "six", "seven", "eight", "nine", "ten" }, "one│ two │three│four│five\nsix│seven│eight│nine│ten \n")]
		public void FormatTable_AlignCenter(String[] firstFormatParams, String firstExpected, String[] secondFormatParams, String secondExpected) {
			FormatTable table = new FormatTable(5, Alignment.Center);
			table.Format(firstFormatParams);
			Assert.Equal(firstExpected, table.ToString().Replace("\r\n", "\n"));
			table.Format(secondFormatParams);
			Assert.Equal(secondExpected, table.ToString().Replace("\r\n", "\n"));
		}
	}
}
